utils/umaApi.ts

```diff
@@ -70,39 +71,90 @@ export type EmpState = {
   identifierPrice: string;
   tokenMarketPrice: string;
-};
-export type EmpStats = {
+  type: "emp";
+}
+interface LspState {
+  id: string;
+  address: string;
+  updated: number;
+  collateralPerPair: string;
+  priceIdentifier: string;
+  collateralToken: string;
+  longToken: string;
+  shortToken: string;
+  finder: string;
+  financialProductLibrary: string;
+  customAncillaryData: string;
+  prepaidProposerReward: number;
+  expirationTimestamp: number;
+  expiryPrice: string;
+  expiryPercentLong: string;
+  contractState: number;
+  totalPositionCollateral: string;
+  sponsors: string[];
+  longTokenDecimals: number;
+  shortTokenDecimals: number;
+  collateralDecimals: number;
+  longTokenName: string;
+  shortTokenName: string;
+  collateralName: string;
+  longTokenSymbol: string;
+  shortTokenSymbol: string;
+  collateralSymbol: string;
+  type: "lsp";
+}
+export type ContractType = "emp" | "lsp";
+export type SynthState<T extends { type: ContractType }> = T extends {
+  type: "emp";
+}
+  ? EmpState
+  : LspState;
+
+export type SynthStats = {
   id: string;
   address: string;
   tvl: string;
   tvm: string;
 };
+export type Synth<T extends { type: ContractType }> = ContentfulSynth &
+  SynthStats &
+  SynthState<T> & { tvl24hChange: number };
+
 export type TimeSeries = {
   id: string;
   address: string;
   value: string;
   timestamp: number;
 }[];
-type GetEmpsAddresses = () => Promise<string[]>;
-type GetEmpState = (address: string) => Promise<EmpState>;
-type GetEmpsState = (address: string) => Promise<EmpState[]>;
-type GetEmpStats = (address: string) => Promise<EmpStats>;
+
+// Function types
+type GetAddresses = () => Promise<string[]>;
+type GetState = <T extends { type: ContractType }>(
+  address: string
+) => Promise<SynthState<T>>;
+
+// type GetSynthsState = (address: string) => Promise<SynthState[]>;
+type GetSynthStats = (address: string) => Promise<SynthStats>;
 type GetStat = (addresses?: string | string[]) => Promise<string>;
 type GetStatBetween = (
   addresses: string | string[],
   startTimestamp?: number
-) => Promise<EmpStats[]>;
+) => Promise<SynthStats[]>;
 
 type GetPriceSlice = (address: string) => Promise<string[]>;
 
-const getEmpsAddresses: GetEmpsAddresses = () => request("listEmpAddresses");
-const getEmpState: GetEmpState = (address) => request("getEmpState", address);
-const getEmpsState: GetEmpsState = () =>
-  getEmpsAddresses().then((addresses) =>
-    Promise.all(addresses.map(getEmpState))
-  );
+const getAddresses: GetAddresses = async () => {
+  const empAddresses: string[] = await request("listEmpAddresses");
+  const lspAddresses: string[] = await request("listAddresses");
+  return [...empAddresses, ...lspAddresses];
+};
+
+const getState: GetState = async (address) => {
+  const state: any = await request("global/getState", address);
+  return state;
+};
 
-const getEmpStats: GetEmpStats = async (address) => {
+const getSynthStats: GetSynthStats = async (address) => {
```

Comment:
i think theres going to be a problem with these calls. The specific calls are not exposed for all types on the global channel.  You need to know the type of the contract to get tvl/tvm from. EMPS have tvm and tvl, whle LSPS only have tvm.  The global channel only has combined tvl of all contracts, not per address, though its possible we can add this